### PR TITLE
[BUG] Add exception handle in MDKeyGenStep

### PR DESCRIPTION
**Pull request by Jay357089**

### CODE REVIEW

**Comment by Jay357089** [comment]
@QiangCai pls check

File: processing/src/main/java/org/carbondata/processing/mdkeygen/MDKeyGenStep.java [hunk 1]
```diff
@@ -188,13 +188,18 @@ public boolean processRow(StepMetaInterface smi, StepDataInterface sdi) throws K
...
+      LOGGER.error(e, "Failed for table: " + this.tableName + " in MDKeyGenStep");
       throw new KettleException("Error while initializing data handler : " + e.getMessage());
+    } catch (Exception e) {
+      LOGGER.error(e, "Failed for table: " + this.tableName + " in MDKeyGenStep");
+      throw new KettleException("There is an unexpected error: " + e.getMessage());
     } finally {
       try {
         dataHandler.finish();
       } catch (CarbonDataWriterException e) {
         LOGGER.debug("Error in  closing data handler ");
```
Comment (QiangCai) [inline processing/src/main/java/org/carbondata/processing/mdkeygen/MDKeyGenStep.java]: I prefer to use error method instead of debug metod

Comment (Jay357089) [inline processing/src/main/java/org/carbondata/processing/mdkeygen/MDKeyGenStep.java]: @QiangCai it is the original code. i don't know why they use LogDebug, but Are you serious that we can change?  @gvramana

Comment (jackylk) [inline processing/src/main/java/org/carbondata/processing/mdkeygen/MDKeyGenStep.java]: I think change to log.error is required. This is not a proper use of debug log

**Comment by Jay357089** [comment]
@QiangCai @jackylk  pls review and check in.

**Pull request merged**